datajunction-query/djqs/engine.py

```diff
@@ -66,22 +67,53 @@ def get_columns_from_description(
     return columns
 
 
+def create_trino_engine(engine: Engine, headers: Optional[Dict[str, str]]) -> "Engine":
+    """
+    Create a SQLAlchemy engine for Trino.
+    """
+    # Parsing the URI
+    parsed_uri = urlparse(engine.uri)
+    scheme = parsed_uri.scheme
+    host = parsed_uri.hostname
+    port = parsed_uri.port
+    path = parsed_uri.path.lstrip("/")
+    catalog, schema = path.split("/", 1) if "/" in path else (path, None)
```

Comment:
Can you explain this string parsing here? What is the format of `engine.uri` you're expecting?

Reply:
I expect the following fields to be available 
```python
ParseResult(scheme='trino', netloc='knorket-abc-trino.kloudsoft.co:443', path='/tpch/sf1', params='', query='', fragment='')
```

Reply:
I see. The string manipulation will be hard to debug when it hits an error. What do you think about making a PosixPath and then just checking the number of parts? So you could replace:
```py
    path = parsed_uri.path.lstrip("/")
    catalog, schema = path.split("/", 1) if "/" in path else (path, None)
```
with:
```py
    from pathlib import PurePosixPath
    ...
    path_parts = PurePosixPath(parsed_uri.path).parts
    if len(path_parts) < 2:
        raise HTTPException(
            status_code=HTTPStatus.BAD_REQUEST,
            detail="Cannot create a Trino engine, catalog or schema missing",
        )
    catalog = path_parts[1] if len(path_parts) > 1 else None
    schema = path_parts[2] if len(path_parts) > 2 else None
```